projects/packages/stats/src/class-wpcom-stats.php

```diff
@@ -0,0 +1,360 @@
+	 *
+	 * @link https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/stats/followers/
+	 * @param array $args Optional query parameters.
+	 * @return array|WP_Error
+	 */
+	public function get_followers( $args = array() ) {
+
+		$this->resource = 'followers';
+
+		return $this->fetch_stats( $args );
+	}
+
+	/**
+	 * Get site's comment followers.
+	 *
+	 * @link https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/stats/comment-followers/
+	 * @param array $args Optional query parameters.
+	 * @return array|WP_Error
+	 */
+	public function get_comment_followers( $args = array() ) {
+
+		$this->resource = 'comment-followers';
+
+		return $this->fetch_stats( $args );
+	}
+
+	/**
+	 * Get site's publicize follower counts.
+	 *
+	 * @link https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/stats/publicize/
+	 * @param array $args Optional query parameters.
+	 * @return array|WP_Error
+	 */
+	public function get_publicize_followers( $args = array() ) {
+
+		$this->resource = 'publicize';
+
+		return $this->fetch_stats( $args );
+	}
+
+	/**
+	 * Get search terms used to find the site.
+	 *
+	 * @link https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/stats/search-terms/
+	 * @param array $args Optional query parameters.
+	 * @return array|WP_Error
+	 */
+	public function get_search_terms( $args = array() ) {
+
+		$this->resource = 'search-terms';
+
+		return $this->fetch_stats( $args );
+	}
+
+	/**
+	 * Get the total number of views for each post.
+	 *
+	 * @link https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/stats/views/posts/
+	 * @param array $args Optional query parameters.
+	 * @return array|WP_Error
+	 */
+	public function get_total_post_views( $args = array() ) {
+
+		$this->resource = 'views/posts';
+
+		return $this->fetch_stats( $args );
+	}
+
+	/**
+	 * Build WPCOM REST API endpoint.
+	 *
+	 * @return string
+	 */
+	protected function build_endpoint() {
+		$resource = ltrim( $this->resource, '/' );
+
+		return sprintf( '/sites/%d/stats/%s', Jetpack_Options::get_option( 'id' ), $resource );
+	}
+
+	/**
+	 * Fetches stats data from WPCOM or local Cache. Caches locally for 5 minutes.
+	 *
+	 * @param array $args Optional query parameters.
+	 *
+	 * @return array|WP_Error
+	 */
+	protected function fetch_stats( $args = array() ) {
+		$endpoint       = $this->build_endpoint();
+		$api_version    = self::STATS_REST_API_VERSION;
+		$cache_key      = md5( implode( '|', array( $endpoint, $api_version, wp_json_encode( $args ) ) ) );
+		$transient_name = self::STATS_CACHE_TRANSIENT_PREFIX . $cache_key;
+		$stats_cache    = get_transient( $transient_name );
+
+		if ( $stats_cache ) {
+			$time = key( $stats_cache );
+			$data = $stats_cache[ $time ]; // JSON encoded data.
+
+			return array_merge( array( 'cached_at' => $time ), (array) json_decode( $data, true ) );
+		}
+
+		$wpcom_stats = $this->fetch_remote_stats( $endpoint, $args );
+
+		if ( is_wp_error( $wpcom_stats ) ) {
```

Comment:
As a note in the [corresponding implementation in the Stats module ](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/stats.php#L1673)we used to cache error responses as well. I think it makes more sense to only cache successful responses

Reply:
Makes sense!